### DISCUSSION TITLE
How should I create knockouts in polygons?

### DISCUSSION THREAD

**Discussion by mathias17** [godotengine/godot-proposals#5428] [2D]
My 2D game project uses large polygons to create ground texture where need be.
One polygon looks like this:
![image](https://user-images.githubusercontent.com/3425436/190491707-c06955dc-0409-4364-bffa-8670acbffe04.png)

The need has arisen to cut holes in these large polygons, creating holes in the ground which the player can see through.
Something like this: (this image is just a photoshopped edit)
![image](https://user-images.githubusercontent.com/3425436/190492362-b8253130-c0e4-4b25-8cc3-0592d1dd696b.png)

How should I go about doing this?

Is using [Goost](https://goost.readthedocs.io/en/gd3/components/geometry.html) recommended?
I don't know if Goost and 4.0 will play nice together.
Goost's polygon booleans seem good:
![image](https://i.imgur.com/FkaoRSD.gif)

**Comment by Zireael07** [comment]
Goost is a completely separate customized Godot editor. I am 90% certain it's based on 3.x branch. 

Old discussion on how to achieve it in Godot itself: https://github.com/godotengine/godot/issues/35676 (Goost just uses an updated, less buggy, version of the library used for polygons, and a slightly different API)

**Comment by theraot** [comment]
[MrElipteach](https://www.youtube.com/channel/UCyNyhJ7QpdeeGwgPTD6Rr0A) has videos on the subject (and without using Goost):

- [SUPER EASY 2D DESTRUCTIBLE TERRAIN in Godot](https://www.youtube.com/watch?v=FiKsyOLacwA)
- [The MOST UNDERRATED Godot tool (GEOMETRY CLASS)](https://www.youtube.com/watch?v=mnDsqD3S1YY)

Addendum: Here is repository he mentions https://github.com/matterda/godot-destructible-terrain
